src/raw/requests.rs

```diff
@@ -14,27 +13,17 @@ use kvproto::tikvpb::TikvClient;
 use std::mem;
 use std::sync::Arc;
 
-pub(super) struct RawGet {
-    pub key: Key,
-    pub cf: Option<ColumnFamily>,
-}
-
-impl KvRequest for RawGet {
+impl KvRequest for kvrpcpb::RawGetRequest {
     type Result = Option<Value>;
-    type RpcRequest = kvrpcpb::RawGetRequest;
     type RpcResponse = kvrpcpb::RawGetResponse;
     type KeyData = Key;
     const REQUEST_NAME: &'static str = "raw_get";
-    const RPC_FN: RpcFnType<Self::RpcRequest, Self::RpcResponse> = TikvClient::raw_get_async_opt;
+    const RPC_FN: RpcFnType<Self, Self::RpcResponse> = TikvClient::raw_get_async_opt;
 
-    fn make_rpc_request<KvC: KvClient>(
-        &self,
-        key: Self::KeyData,
-        store: &Store<KvC>,
-    ) -> Self::RpcRequest {
-        let mut req = store.request::<Self::RpcRequest>();
+    fn make_rpc_request<KvC: KvClient>(&self, key: Self::KeyData, store: &Store<KvC>) -> Self {
```

Comment:
We don't need this method any more, we can have a `set_store` method instead

Reply:
We still need to know how to set `KeyData` because they can be split to smaller parts. And it's also not so convenient to fill other option fields with reused code (for example, CF is a String which is not Copy).

Reply:
I don't understand - I thought we set the key data and family when we create the request, and so here all we need to do is set the store/context (and maybe clone).

Reply:
`make_rpc_request` is now used in the `response_stream` function when a small part of key data is generated. So we must know how to set key data at least.